plugin/trino-base-jdbc/src/test/java/io/trino/plugin/jdbc/TestReusableConnectionFactory.java

```diff
@@ -153,6 +154,26 @@ public void testConnectionIsNotReusedWhenAutoCommitDisabled()
         }
     }
 
+    @Test
+    public void testCachedConnectionWillRefreshWhenItsClosed()
+            throws Exception
+    {
+        MockConnectionFactory mockConnectionFactory = new MockConnectionFactory();
+        ReusableConnectionFactory connectionFactory = new ReusableConnectionFactory(mockConnectionFactory, FOREVER, HUGE_SIZE);
+        Connection connection = connectionFactory.openConnection(ALICE);
+
+        if (connection instanceof ReusableConnectionFactory.CachedConnection cachedConnection) {
+            cachedConnection.delegate().close();
+            assertThat(cachedConnection.isClosed()).isFalse();
+            assertThat(cachedConnection.delegate().isClosed()).isFalse();
+        }
+        else {
+            fail("Acquired connection is not cached");
+        }
+        assertThat(mockConnectionFactory.openedConnections).isEqualTo(2);
+        assertThat(mockConnectionFactory.closedConnections).isEqualTo(1);
```

Comment:
Can you call `connectionFactory.openConnection(ALICE)` to make sure the new connection was created?